environs/tools/build.go

```diff
@@ -301,23 +301,43 @@ func packageLocalTools(toolsDir string, buildAgent bool) error {
 type BundleToolsFunc func(build bool, w io.Writer, forceVersion *version.Number) (version.Binary, bool, string, error)
 
 // Override for testing.
-var BundleTools BundleToolsFunc = bundleTools
+var BundleTools BundleToolsFunc = func(
+	build bool, w io.Writer, forceVersion *version.Number,
+) (version.Binary, bool, string, error) {
+	return bundleTools(build, w, forceVersion, JujudVersion)
+}
 
 // bundleTools bundles all the current juju tools in gzipped tar
 // format to the given writer.  If forceVersion is not nil and the
 // file isn't an official build, a FORCE-VERSION file is included in
 // the tools bundle so it will lie about its current version number.
-func bundleTools(build bool, w io.Writer, forceVersion *version.Number) (_ version.Binary, official bool, sha256hash string, _ error) {
+func bundleTools(
+	build bool, w io.Writer, forceVersion *version.Number, jujudVersion func(dir string) (version.Binary, bool, error),
+) (_ version.Binary, official bool, sha256hash string, _ error) {
 	dir, err := ioutil.TempDir("", "juju-tools")
 	if err != nil {
 		return version.Binary{}, false, "", err
 	}
 	defer os.RemoveAll(dir)
+
+	existingJujuLocation, err := ExistingJujuLocation()
+	if err != nil {
+		return version.Binary{}, false, "", errors.Annotate(err, "couldn't find existing jujud")
+	}
```

Comment:
If build is specified and there's no existing jujud, we don't want to error

Reply:
the location here is the juju client but not jujud